### PR TITLE
Spacelube slide

### PR DESCRIPTION
**Pull request by Arendian**
## About the PR

When you slip on space lube you now lose all friction and will keep sliding until you reach the end of the puddle.

## Why / Balance

Large puddles of lube  are annoying to traverse, you can only move maybe 1 tile before you fall down again. Now you just fall down once and slide to the other side.
Now that entities keep sliding until they reach the end of the puddle, you can make lube traps that slide people into ~~dangerous~~ fun situations.

## Technical details

Puddles can now be SuperSlippery, if an entity slips on a SuperSlippery puddle, they will get the SlidingComponent. While an entity has the SlidingComponent, any friction event calculations will be multiplied by 0(no friction). When an entity with the SlidingComponent stops coliding with a SuperSlippery puddle, the friction multiplier will be set to the one that entities normally get when knocked down. When the entity stands up again, the entity loses the SlidingComponent.
SlidingComponent prevents further velocity increases from other slips.

## Media

The stutter at 19 seconds was a mispredict that should be fixed now.

https://github.com/space-wizards/space-station-14/assets/137322659/b705c26f-81fa-49e7-8989-5993204fa524

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: Dygon
- tweak: When slipping on lube you now keep sliding until you reach a tile without lube.

### CODE REVIEW

**Comment by Hmeister-real** [comment]
Best PR of 2024 so far, second to eating paper.

**Comment by Golinth** [comment]
how does this interact with conveyor belts? I'm worried that it could be made into an unavoidable stunlock/forced ride somewhere

File: Content.Shared/Slippery/SlidingSystem.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    private void OnEndCollide(EntityUid uid, SlidingComponent component, ref EndCollideEvent args)
+    {
+        if (TryComp<SlipperyComponent>(args.OtherEntity, out var slippery) && slippery.SuperSlippery)
+            component.CollidingEntities.Remove(args.OtherEntity);
+
+        if (component.CollidingEntities.Count == 0)
+            component.FrictionModifier = SharedStunSystem.KnockDownModifier;
+    }
+
+}
```
Comment (metalgearsloth) [inline Content.Shared/Slippery/SlidingSystem.cs:60]: 1. These need networking with dirty calls.
2. You should just remove on endcollide as this has a chance to leak if the component disappears, then avoid checking the count if nothing changed.

**Comment by deltanedas** [comment]
sseth referen

**Pull request merged**